src/Task3Test.kt

```diff
@@ -7,17 +7,17 @@ internal class Task3Test {
     //Exceptional case where index -1 is returned
     // to show that an array of length of less than
     // 2 was given as input.
-    fun onEmptyArray() = assertTrue(!intArrayOf().hasDuplicates())
+    fun onEmptyArray() = assertTrue(!hasDuplicates(intArrayOf()))
 
     @Test
     // Same bad input scenario.
-    fun onSingleElementArray() = assertTrue(!intArrayOf().hasDuplicates())
+    fun onSingleElementArray() = assertTrue(!hasDuplicates(intArrayOf(1)))
 
     @Test
-    //Usual scenario.
-    fun onNoDuplicatesArrayOfInt() = assertTrue(intArrayOf(1,2,3,4,5,6,7,8,9,10).hasDuplicates())
+    //Usual scenario without duplicates.
+    fun onNoDuplicatesArrayOfInt() = assertTrue(hasDuplicates(intArrayOf(1,2,3,4,5,6,7,8,9,10)))
```

Comment:
Well actually this array has no duplicates.

Your solution seems promising but is wrong.

Reply:
A bug in code backed up by a bug in the test.. 👍 Fixed